consensus/istanbul/validator/default.go

```diff
@@ -358,8 +358,12 @@ func (valSet *defaultSet) Copy() istanbul.ValidatorSet {
 	}
 
 	newValSet := NewSubSet(addresses, valSet.policy, valSet.subSize).(*defaultSet)
-	_, proposer := newValSet.GetByAddress(valSet.GetProposer().Address())
-	newValSet.proposer.Store(proposer)
+	if proposer := newValSet.GetProposer(); proposer != nil {
+		_, proposer := newValSet.GetByAddress(proposer.Address())
+		newValSet.proposer.Store(proposer)
+	} else {
+		logger.Error("Found nil proposer", "validators", len(valSet.validators))
+	}
```

Comment:
@hyunsooda In the known issues(https://github.com/klaytn/klaytn/issues/1206, https://github.com/klaytn/klaytn/issues/1612), The panic is caused because `NewValSet` does not contain original valSet's proposer (`valSet.proposer`). 

So, What about next code?
```
	idx, proposer := newValSet.GetByAddress(valSet.GetProposer().Address())
	if idx != -1 {
		newValSet.proposer.Store(proposer)
	}
	// The original proposer is removed but new proposer is not calculated
	// because it's syncing. Set default proposer as a proposer.
```
cc. I think setting random proposer as a newValSet.proposer works well because if the node enters mining mode, it calculates the original proposer anyway.. @jiseongnoh

Reply:
Your suggestion may still cause panic because `GetProposer()` may return nil and the code suggested does not have a guard whether it is nil or not.

Reply:
@hyunsooda Ok, then i think adding valSet.proposer nil checking logic. Was this what you pointed out? 
```
	if proposer := valSet.GetProposer(); proposer != nil {
		if idx, _ := newValSet.GetByAddress(proposer.Address()); idx != -1 {
			newValSet.proposer.Store(proposer)
		}
	}

	// The original proposer is removed but new proposer is not calculated
	// because it's syncing. Set default proposer as a proposer.
```

Reply:
btw, can i ask why you check,set,and logerror the newValSet.proposer here? because it is already set at NewSubSet.

Reply:
The originally proposed changes in `backend` package were rollbacked since those checks are unnecessary.
 
The panic by snapshot error was hard to reproduce because its context is too less. I think it’s better to wait until a new report is issued.

The panic is not observable now, but let me note one thing here that maybe you already know. The released validators cannot request block sync to current validators. To do that, (1) running `en` binary works once the existing chaindata migration is done. Also, (2) the `cn` binary can still sync with `en` process via peer connection without chaindata migration.

I found a vulnerability and posted its description. [You can see here](https://github.com/klaytn/klaytn/issues/1690).

Reply:
@hyunsooda Could you add `TODO` about this context?